test/test_cost.py

```diff
@@ -10332,7 +10332,9 @@ def test_dreamer_actor(self, device, imagination_horizon, discount_loss, td_est)
             return
         if td_est is not None:
             loss_module.make_value_estimator(td_est)
-        loss_td, fake_data = loss_module(tensordict)
+        # NOTE: Input is reshaped because GRUCell (which is part of the
+        # RSSMPrior module in `mb_env`) expects input to be either 1D or 2D
+        loss_td, fake_data = loss_module(tensordict.reshape(-1))
```

Comment:
I'm not sure if there is a better way to fix this. I suppose it could be possible to just reshape the direct input to the GRUCell? But I'm not sure how to do that

Reply:
if we need to reshape we should reshape - but another option here would be to use vmap
like:
```python
if tensordict.ndim > 1:
    loss_td, fake_data = vmap(loss_module, (0,))(tensordict)
```
(gru works with vmap as long as you are using the python only version in torchrl.modules)

Reply:
If I try using `VmapModule`, I get this error:

```
  File "/home/endoplasm/develop/torchrl-1/test/test_cost.py", line 10338, in test_dreamer_actor
    loss_td, fake_data = VmapModule(loss_module, (0,))(tensordict)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/endoplasm/develop/torchrl-1/torchrl/modules/tensordict_module/common.py", line 454, in __init__
    self.in_keys = module.in_keys
                   ^^^^^^^^^^^^^^
  File "/home/endoplasm/develop/torchrl-1/torchrl/objectives/common.py", line 441, in __getattr__
    return super().__getattr__(item)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/endoplasm/miniconda/envs/torchrl-1/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1931, in __getattr__
    raise AttributeError(
AttributeError: 'DreamerActorLoss' object has no attribute 'in_keys'
```

I'll probably just leave the reshape for now, but I would like to understand this.

Indeed if I try to access `loss_module.in_keys` directly, I also get the above error. But I can access the `in_keys` of the actor model and world model within the loss module:
```
print(loss_module.actor_model.in_keys)
print(loss_module.model_based_env.world_model.in_keys)
```
```
['state', 'belief']
['state', 'belief', 'action']
```

So I'm wondering what would be the right way to make `VmapModule` and `DreamerActorLoss` compatible?

Reply:
Oh yeah `DreamerActorLoss` should have in_keys!
All losses should. Dreamer hasn't received much love lately as you can see. 
Let's take care of that in a separate PR then